load_balancer/test/integration/load_balancerTest.js

```diff
@@ -0,0 +1,243 @@
+console.log('======integration tests on load balancer======\n');
+
+const chai = require('chai');
+const dirtyChai = require('dirty-chai');
+const request = require('request');
+const nock = require('nock');
+const dns = require('dns');
+const sinon = require('sinon');
+const { check } = require('../../../util/environmentCheck.js');
+const testData = require('./data/submission.json');
+// eslint-disable-next-line import/no-dynamic-require
+const configData = require(`../../${process.env.LBCONFIG}`);
+const lbUrl = `https://${configData.load_balancer.hostname}:${configData.load_balancer.port}`;
+const sandbox = sinon.createSandbox();
+
+let loadBalancer;
+let logStub;
+
+check('LBCONFIG');
+
+chai.should();
+chai.use(dirtyChai);
+
+const activateNocks = function activateNocks() {
+  if (!nock.isActive()) {
+    nock.activate();
+  }
+};
+
+const cleanNocks = function cleanNocks() {
+  nock.cleanAll();
+  nock.restore();
+};
+
+const stubConsole = function stubConsole() {
+  logStub = sandbox.stub(console, 'log');
+};
+
+const restoreConsole = function restoreConsole() {
+  logStub = {};
+  sandbox.restore();
+};
+
+const startLoadBalancer = function startLoadBalancer() {
+  delete require.cache[require.resolve('../../load_balancer.js')];
+  // eslint-disable-next-line global-require
+  loadBalancer = require('../../load_balancer.js');
+};
+
+const connectionCheckNock = function connectionCheckNock(url, returnVal) {
+  const nockObj = nock(url)
+    .get('/connectionCheck')
+    .reply(200, (uri) => returnVal);
+  return nockObj;
+};
+
+const requestRunNock = function requestRunNock(url) {
+  const tempJob = testData.submissionsData[0];
+  const nockObj = nock(url)
+    .persist()
+    .post('/requestRun')
+    .reply(200, (uri, requestBody) => {
+      requestBody.should.have.all.keys('id_no', 'Lab_No', 'time', 'commit', 'status', 'penalty', 'socket', 'language');
+      requestBody.should.deep.equal(tempJob);
+      return true;
+    });
+  return nockObj;
+};
+
+const connectionCheckAssert = function connectionCheckAssert(upArray, resp) {
+  request.get(`${lbUrl}/connectionCheck`, (error, response) => {
+    const responseBody = JSON.parse(response.body);
+    responseBody.components.forEach((elem, index) => {
+      if (upArray.includes(index)) {
+        elem.status.should.equal('up');
+      } else {
+        elem.status.should.equal('down');
+      }
+    });
+    restoreConsole();
+    resp();
+  });
+};
+
+const submitAssert = function submitAssert() {
```

Comment:
Please improve the name to something more appropriate.

Reply:
Sir, how about jobSubmitCheck?